docs/wiki/EL/el-architecture.md

```diff
@@ -1,66 +1,279 @@
 
-# Transaction pools
+**JSON-RPC API**
 
-we have primarily two kinds of txPools :
+When utilizing a wallet or a DApp, our communication with the execution layer is conducted over a standardized JSON-RPC API. This enables us to externally query the Ethereum state or dispatch a transaction to it, signed by the wallet, which is subsequently validated by the execution layer client and disseminated around the network.
 
-1. legacy pools- this in geth contains price-sorted heaps / priority-queues
-   where the transactions are sorted by price
-   Two heaps are used for sorting: one is based on the effective tip in the next block) and the other based on gasFeeCap. On saturation the bigger of the two is chosen for eviction
-2. blob pools - this also keeps a priority heap for eviction but there are a few things different about this, the code has an excelent comments section for these [blobPools](https://github.com/ethereum/go-ethereum/blob/064f37d6f67a012eea0bf8d410346fb1684004b4/core/txpool/blobpool/blobpool.go#L132) also note that logarithms are used in eviction queues in blob pools
+**Engine API**
 
-In Ethereum two primary types of transaction pools (txPools) are recognized:
+This is the only link between the consensus and execution layer. The engine exposes two major classes of endpoints to the consensus layer: **fork choice updated** and **new payload** suffixed by the three versions they are exposed as (V1-V3).These methods encapsulate two major pipelines offered by the execution layer:
 
-1. **Legacy Pools**: Managed by Geth, these pools employ price-sorted heaps or priority queues to organize transactions based on their price. Specifically, transactions are arranged using two heaps: one prioritizes the effective tip for the upcoming block, and the other focuses on the gas fee cap. During periods of saturation, the larger of these two heaps is selected for the eviction of transactions, optimizing the pool's efficiency and responsiveness. [urgent and floating heaps](https://github.com/ethereum/go-ethereum/blob/064f37d6f67a012eea0bf8d410346fb1684004b4/core/txpool/legacypool/list.go#L525)
+1.  **New Payload\***(V1/V2/V3)\*: payload validation & insertion pipeline.
+2.  **Fork Choice Updated\***(V1/V2/V3)\*: state synchronization & block building pipeline.
+
+**Sync**
+
+In order to accurately process transactions on Ethereum, it is imperative that we reach a consensus on the global status of the network, rather than solely relying on our local perspective. The global state synchronization of the execution layer client is triggered by the fork choice rule governed by the LMD-GHOST algorithm in the consensus layer. It is then relayed to the execution layer through the fork choice updated endpoint of the engine API. Syncing entails two possible processes: downloading remote blocks from peers and validating them in the EVM.
+
+Note: client specific overviews will go here
+
+### Reth
```

Comment:
The part dedicated to reth is pretty client specific and thorough. It could become its own page similar to #191 with extra info for contributors and make this page leaner.

Reply:
Sounds great, will do that at some point today